kernel/src/expressions/mod.rs

```diff
@@ -36,12 +36,39 @@ pub enum BinaryOperator {
     Distinct,
 }
 
+impl BinaryOperator {
+    // invert an operator. Returns Some<InvertedOp> if the operator supports inversion, None if it
+    // cannot be inverted
+    pub(crate) fn invert(&self) -> Option<BinaryOperator> {
```

Comment:
Are you just thinking that some of the data_skipping expr construction could be simplified with `commute`? We do need to commute there, but we also change the actual expressions (i.e. the column name becomes `stats.maxValues.colName` or similar) so since we have to rebuild anyway I'm not sure `commute` is needed?

Reply:
Done